src/snakemake/scheduling/job_scheduler.py

```
@@ -3,27 +3,31 @@
 __email__ = "johannes.koester@uni-due.de"
 __license__ = "MIT"
 
+import asyncio
 from bisect import bisect
 from collections import defaultdict, deque
```

Ruff (0.12.2)

`collections.defaultdict` imported but unused

Remove unused import: `collections.defaultdict`

(F401)